pyteck/simulation.py

```diff
@@ -207,60 +210,135 @@ def __init__(self, kind, apparatus, meta, properties):
+        elif self.composition_type == 'mass fraction':
             self.gas.TPY = temp, pres, reactants
         else:
             raise(BaseException('error: not supported'))
             return
+        return self.gas
+
+    def run_case(self,restart=False):
+        if restart and os.path.isfile(self.meta['save-file']):
+            print('Skipped existing case ', self.meta['id'])
+            return
+
+        # Save simulation results in hdf5 table format.
+        table_def = {'time': tables.Float64Col(pos=0),
+                     'temperature': tables.Float64Col(pos=1),
+                     'pressure': tables.Float64Col(pos=2),
+                     'volume': tables.Float64Col(pos=3),
+                     }
+        if self.kind =='ignition delay':
+            table_def['mass_fractions']=tables.Float64Col(
+                          shape=(self.reac.thermo.n_species), pos=4
+                          )
+        elif self.kind == 'species profile':
+            table_def['mole_fractions']=tables.Float64Col(
+                          shape=(self.reac.thermo.n_species), pos=4
+                          )
+
+        with tables.open_file(self.meta['save-file'], mode='w',
+                              title=self.meta['id']
+                              ) as h5file:
+
+            table = h5file.create_table(where=h5file.root,
+                                        name='simulation',
+                                        description=table_def
+                                        )
+            # Row instance to save timestep information to
+            timestep = table.row
+            # Save initial conditions
+            timestep['time'] = self.reac_net.time
+            timestep['temperature'] = self.reac.T
+            timestep['pressure'] = self.reac.thermo.P
+            timestep['volume'] = self.reac.volume
+            if self.kind == 'ignition delay':
+                timestep['mass_fractions'] = self.reac.Y
+            elif self.kind == 'species profile':
+                timestep['mole_fractions'] = self.reac.thermo.X
+            # Add ``timestep`` to table
+            timestep.append()
+
+            # Main time integration loop; continue integration while time of
+            # the ``ReactorNet`` is less than specified end time.
+            while self.reac_net.time < self.time_end:
+                self.reac_net.step()
+
+                # Save new timestep information
+                timestep['time'] = self.reac_net.time
+                timestep['temperature'] = self.reac.T
+                timestep['pressure'] = self.reac.thermo.P
+                timestep['volume'] = self.reac.volume
+                if self.kind == 'ignition delay':
+                    timestep['mass_fractions'] = self.reac.Y
+                elif self.kind == 'species profile':
+                    timestep['mole_fractions'] = self.reac.thermo.X
+                # Add ``timestep`` to table
+                timestep.append()
```

Comment:
JSR simulations needn't save time history.
Probably this whole method should be in the AutoIgnitionSimulation class, not the Simulation super-class